data/src/client.rs

```diff
@@ -108,7 +109,14 @@ impl Client {
             let user = config.username.as_ref().unwrap_or(nick);
             let real = config.realname.as_ref().unwrap_or(nick);
 
-            if let Some(pass) = config.password.as_ref() {
+            if let Some(pass_file) = config.password_file.as_ref() {
+                match fs::read_to_string(pass_file) {
+                    Ok(pass) => {
```

Comment:
We shouldn't block here as this is all run on the main UI thread. Instead, we should load the password into memory at our entry point in main and pass it in. This can be done right after we load the config file.

Reply:
Moved the file reading to Config::load();